src/main/java/com/cleanroommc/groovyscript/mapper/BlockStateMapper.java

```diff
@@ -0,0 +1,49 @@
+package com.cleanroommc.groovyscript.mapper;
+
+import com.cleanroommc.groovyscript.api.Result;
+import com.cleanroommc.groovyscript.compat.mods.GroovyContainer;
+import com.cleanroommc.groovyscript.server.Completions;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.common.registry.ForgeRegistries;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+
+public class BlockStateMapper extends AbstractObjectMapper<IBlockState> {
+
+    public static final BlockStateMapper INSTANCE = new BlockStateMapper("blockstate", null);
+
+    protected BlockStateMapper(String name, GroovyContainer<?> mod) {
+        super(name, mod, IBlockState.class);
+    }
+
+    @Override
+    public Result<IBlockState> getDefaultValue() {
+        return Result.some(Blocks.AIR.getDefaultState());
+    }
+
+    @Override
+    public @NotNull Result<IBlockState> parse(String mainArg, Object[] args) {
```

Comment:
this and a few other places have switched from `parse(String mainArg, Object... args)` to `parse(String mainArg, Object[] args)` - while this makes no difference to groovy (or to overrides, apparently, which is interesting), is this intentional?
should we continue using varargs for things exposed to groovy, even if it makes no difference in functionality (although it does display differently on the wiki)?